### PR TITLE
Run unit tests in selected modules concurrently

### PR DESCRIPTION
**Pull request by snazy**
Use `junit-platform-maven-plugin` instead of `maven-surefire-plugin` for selected modules,
enable parallel (class level) execution by default, enable concurrent (method level) execution
where appropriate.

Without this change (from CI):
```
[INFO] Nessie - Model ..................................... SUCCESS [ 44.331 s]
[INFO] Nessie - Client .................................... SUCCESS [ 40.998 s]
[INFO] Nessie - Services .................................. SUCCESS [ 23.074 s]
[INFO] Nessie - Versioned Store SPI ....................... SUCCESS [ 29.017 s]
```

With this change (from CI):
```
[INFO] Nessie - Model ..................................... SUCCESS [ 29.581 s]
[INFO] Nessie - Client .................................... SUCCESS [ 22.951 s]
[INFO] Nessie - Services .................................. SUCCESS [ 15.026 s]
[INFO] Nessie - Versioned Store SPI ....................... SUCCESS [ 17.495 s]
```

### CODE REVIEW

File: model/src/test/java/org/projectnessie/error/TestNessieError.java [hunk 1]
```diff
@@ -59,7 +62,7 @@ void fullMessage() {
                 + "foo.bar.InternalServerError\n"
                 + "\tat some.other.Class\n"
                 + "java.lang.Exception: processingException\n"
-                + "\tat org.projectnessie.error.TestNessieError.fullMessage(TestNessieError.java:");
```
Comment (dimas-b) [inline model/src/test/java/org/projectnessie/error/TestNessieError.java:62]: Is this change needed for concurrent runs? why?

**Review by dimas-b — COMMENTED** [review]
LGTM :+1: just a couple of minor comments

File: pom.xml [hunk 2]
```diff
@@ -764,6 +764,30 @@
         </plugin>
 
         <!-- 3rd party plugins -->
+        <plugin>
+          <groupId>de.sormuras.junit</groupId>
+          <artifactId>junit-platform-maven-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (snazy) [reply pom.xml:769]: Maybe not a commit, but a comment ;)

File: model/src/test/java/org/projectnessie/error/TestNessieError.java (hunk 1 above)
Comment (dimas-b) [reply model/src/test/java/org/projectnessie/error/TestNessieError.java:62]: The test original intention was to ensure that the client-side stack trace is reflected in the full message. Could you add another (more relaxed) check to validate that `TestNessieError.fullMessage` is there?

Comment (snazy) [reply model/src/test/java/org/projectnessie/error/TestNessieError.java:62]: ah, you're right. gimme a sec

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply pom.xml:772]: ninja edited commit -> comment ;-)

File: model/src/test/java/org/projectnessie/error/TestNessieError.java (hunk 1 above)
Comment (snazy) [reply model/src/test/java/org/projectnessie/error/TestNessieError.java:62]: done

**Review by dimas-b — APPROVED** [review]
